builder/builder.go

```diff
@@ -511,46 +526,44 @@ func (b *Builder) convertAndBuild(ctx context.Context, target domain.Target, opt
 					console.Printf("Did not push %s. Use earthly --push to enable pushing\n", saveImage.DockerTag)
 				}
 			}
-			if !sts.Target.IsRemote() {
-				for _, saveLocal := range sts.SaveLocals {
-					artifactDir := filepath.Join(outDir, fmt.Sprintf("index-%d", dirIndex))
-					artifact := domain.Artifact{
-						Target:   sts.Target,
-						Artifact: saveLocal.ArtifactPath,
-					}
-					err := b.saveArtifactLocally(ctx, artifact, artifactDir, saveLocal.DestPath, sts.ID, opt, saveLocal.IfExists)
-					if err != nil {
-						return nil, err
-					}
-					dirIndex++
+			for _, saveLocal := range sts.SaveLocals {
```

Comment:
Shouldn't the `!sts.Target.IsRemote()` still be checked somewhere for 0.5 behavior?

Reply:
I needed to remove it to support the case of performing a build such as:

    earthly github.com/earthly/earthly:main+deps

which would then save the `go.mod` and `go.sum` files locally; however if the target is remote, we perform this extra check earlier up:

```
	if target.IsRemote() {
		if len(featureFlagOverrides) != 0 {
			featureFlagOverrides += ","
		}
		featureFlagOverrides += "referenced-save-only"
	}
```

which will prevent the remote target from saving anything else that was indirectly referenced.

I was thinking that in this case it's fair to assume that if the user is directly referencing a remote target, then that user wants any directly referenced files to be saved.

There might be an issue where a user gives us a local Earthfile that then references something that's remote, and none of these Earthfiles have any VERSION flags, then those remote targets might also be saved -- let me double check this case (and write a test for it if one doesn't exist).

Reply:
In addition this also had to be added to https://github.com/earthly/earthly/blob/selective-push/earthfile2llb/converter.go#L1199